spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,225 @@
+            else:
+                self.set_probe(probe, in_place=True)
+
+        # set channel properties
+        # sometimes there are missing metadata files on the IBL side
+        # when this happens a statement is printed to stderr saying these are using default metadata configurations
+        with redirect_stderr(StringIO()):
+            electrodes_geometry = self._file_streamer.geometry
+
+        if not load_sync_channel:
+            shank = electrodes_geometry["shank"]
+            shank_row = electrodes_geometry["row"]
+            shank_col = electrodes_geometry["col"]
+            inter_sample_shift = electrodes_geometry["sample_shift"]
+            adc = electrodes_geometry["adc"]
+            index_on_probe = electrodes_geometry["ind"]
+            good_channel = electrodes_geometry["flag"]
+        else:
+            shank = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            shank_row = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            shank_col = np.concatenate((electrodes_geometry["shank"], [np.nan]))
+            inter_sample_shift = np.concatenate((electrodes_geometry["sample_shift"], [np.nan]))
+            adc = np.concatenate((electrodes_geometry["adc"], [np.nan]))
+            index_on_probe = np.concatenate((electrodes_geometry["ind"], [np.nan]))
+            good_channel = np.concatenate((electrodes_geometry["shank"], [1.]))
+
+        self.set_property("shank", shank)
+        self.set_property("shank_row", shank_row)
+        self.set_property("shank_col", shank_col)
+        self.set_property("inter_sample_shift", inter_sample_shift)
+        self.set_property("adc", adc)
+        self.set_property("index_on_probe", index_on_probe)
+        if not all(good_channel):
+            self.set_property("good_channel", good_channel)
+
+        # init recording segment
+        recording_segment = StreamingIblRecordingSegment(
+            file_streamer=self._file_streamer, load_sync_channel=load_sync_channel
+        )
+        self.add_recording_segment(recording_segment)
+
+        self._kwargs = {
+            "session": session,
+            "stream_name": stream_name,
+            "load_sync_channel": load_sync_channel,
+            "cache_folder": cache_folder,
+            "remove_cached": remove_cached,
+        }
+
+
+class StreamingIblRecordingSegment(BaseRecordingSegment):
+    def __init__(self, file_streamer, load_sync_channel: bool = False):
+        BaseRecordingSegment.__init__(self, sampling_frequency=file_streamer.fs)
+        self._file_streamer = file_streamer
+        self._load_sync_channel = load_sync_channel
+
+    def get_num_samples(self):
+        return self._file_streamer.ns
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        traces = self._file_streamer[start_frame:end_frame]
```

Comment:
@CodyCBakerPhD looking at some traces I suspect that they are already in Volts. 

There are 2 options here:

1. we convert back to int16 by doing `traces * 1e6 / gain_to_uV`
2. we convert to uV and drop the `gain_to_uV` (otherwise we would convert twice). We can still keep the gain property as `original_gain`

Reply:
I vote for option 1, so we don't do additional operations on the fly. Note that the traces will always be in uV in this case.

Reply:
I pushed option 1 :)

Reply:
and opened an issue on the ibl-lib side: https://github.com/int-brain-lab/ibllib/issues/560

Reply:
@alejoe91 It looks as though you pushed option 2 (keeping as `float32`, just adjusting from V to uV)

So I'm intending to write this data to NWB

A quick test comparing writing the data as `int16` vs. `float32` shows about a 33% increase in disk space when writing as `float32` (compression enabled in both cases; higher in the `float32` case but not 2x as good as `int16`)

I'm planning to write an estimated 62 TB worth of this to start with, so I'd rather that not take up more space than needed

It also looks like the chunk that gets downloaded on the fly to the cache is in the original `int16` type, so it's just the API that scales automatically

Could we maybe add a `return_scaled` option to this `get_traces`, and if `True` do what you've implemented, and if `False` (which I would use) actually do option 1, cast and return as `int16`?

Reply:
Yep sorry I meant option 2. 

Ok for me to do option 1 and option 2 (in uV) when `return_scaled` is True. The optimal solution would be that the `Streamer` gets updated accordingly!

Reply:
> Ok for me to do option 1 and option 2 (in uV) when return_scaled is True. 

Done


> The optimal solution would be that the Streamer gets updated accordingly!

Agreed

Reply:
I also renamed the temporary property to make it clearer what it's supposed to do. If they adjust the return behavior then we can return the class gain/offset properties to normal SpikeInterface behavior